doc/lua_api.txt

```diff
@@ -2896,6 +2896,25 @@ For the following functions `x` can be either a vector or a number:
 * `vector.divide(v, x)`:
     * Returns a scaled vector or Schur quotient.
 
+For the following functions `a` is an angle in radians and `r` is a rotation
+vector ({x = <pitch>, y = <yaw>, z = <roll>}) where pitch, yaw and roll are
+in radians.
+
+* `vector.rotate(v, r):
+    * Applies the rotation `r` to `v` and returns the result.
+* `vector.rotate_around_axis(v1, v2, a):
+    * Returns `v1` rotated around axis `v2` by `a` radians according to
+      the left hand rule.
+* `vector.forward_at_rotation(r)`:
+    * Returns a unit vector pointing forward ({x = 0, y = 0, z = 1}) rotated by `r`.
+    * vector.forward_at_rotation(object:get_rotation()) points forward relative to an entity.
+* `vector.up_at_rotation(r)`:
+    * Returns a unit vector pointing up ({x = 0, y = 1, z = 0}) rotated by `r`.
+    * vector.up_at_rotation(object:get_rotation()) points up relative to an entity.
+* `vector.directions_to_rotation(v1, v2):
+    * Returns a rotation vector for `v1` pointing forward and `v2` pointing up.
```

Comment:
I feel like this needs a better description, because I don't get it.

Reply:
so, a direction by itself cannot be resolved to a single rotation - there are many roll rotations that match the same direction. Think a plane moving in the same direction but spinning. The up vector is needed to resolve that.

I'd prefer the arguments to be named better: `forward` and `up`.